google-cloud-spanner/src/test/java/com/google/cloud/spanner/OpenTelemetryBuiltInMetricsTracerTest.java

```diff
@@ -345,6 +363,58 @@ public void testNoNetworkConnection() {
         1, getAggregatedValue(attemptCountMetricData, expectedAttributesCreateSessionFailed));
   }
 
+  @Test
+  public void testNoServerTimingHeader() throws IOException, InterruptedException {
+    // Create Spanner Object without headers
+    InetSocketAddress addressNoHeader = new InetSocketAddress("localhost", 0);
+    Server serverNoHeader =
+        NettyServerBuilder.forAddress(addressNoHeader).addService(mockSpanner).build().start();
+    String endpoint = address.getHostString() + ":" + serverNoHeader.getPort();
+    Spanner spannerNoHeader =
+        SpannerOptions.newBuilder()
+            .setProjectId("test-project")
+            .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
+            .setHost("http://" + endpoint)
+            .setCredentials(NoCredentials.getInstance())
+            .setSessionPoolOption(
+                SessionPoolOptions.newBuilder()
+                    .setWaitForMinSessionsDuration(Duration.ofSeconds(5L))
+                    .setFailOnSessionLeak()
+                    .setSkipVerifyingBeginTransactionForMuxRW(true)
+                    .build())
+            // Setting this to false so that Spanner Options does not register Metrics Tracer
+            // factory again.
+            .setBuiltInMetricsEnabled(false)
+            .setApiTracerFactory(metricsTracerFactory)
+            .build()
+            .getService();
+    DatabaseClient databaseClientNoHeader =
+        spannerNoHeader.getDatabaseClient(DatabaseId.of("test-project", "i", "d"));
+
+    databaseClientNoHeader
+        .readWriteTransaction()
+        .run(transaction -> transaction.executeUpdate(UPDATE_RANDOM));
+
+    Attributes expectedAttributes =
+        expectedCommonBaseAttributes
+            .toBuilder()
+            .putAll(expectedCommonRequestAttributes)
+            .put(BuiltInMetricsConstant.STATUS_KEY, "OK")
+            .put(BuiltInMetricsConstant.METHOD_KEY, "Spanner.ExecuteSql")
+            .build();
+
+    MetricData gfeConnectivityMetricData =
+        getMetricData(metricReader, BuiltInMetricsConstant.GFE_CONNECTIVITY_ERROR_NAME);
+    assertThat(getAggregatedValue(gfeConnectivityMetricData, expectedAttributes)).isEqualTo(1);
+
+    MetricData afeConnectivityMetricData =
+        getMetricData(metricReader, BuiltInMetricsConstant.AFE_CONNECTIVITY_ERROR_NAME);
+    assertThat(getAggregatedValue(afeConnectivityMetricData, expectedAttributes)).isEqualTo(1);
```

Comment:
AFE header feature at this stage should be off by default and the expected value should be 0?

Reply:
Yes, initially I made it enabled by default as per the plan. Will make it default OFF and push the changes